hyundai_kia_connect_api/KiaUvoApiEU.py

```diff
@@ -826,12 +917,136 @@ def _get_charge_limits(self, token: Token, vehicle: Vehicle) -> dict:
             url, headers=self._get_authenticated_headers(token)
         ).json()
         _LOGGER.debug(f"{DOMAIN} - Get Charging Limits Response: {response}")
-        _check_response_for_errors()
+        _check_response_for_errors(response)
         # API sometimes returns multiple entries per plug type and they conflict.
         # The car itself says the last entry per plug type is the truth when tested (EU Ioniq Electric Facelift MY 2019)
         if response["resMsg"] is not None:
             return response["resMsg"]
 
+    def _get_trip_info(
+        self,
+        token: Token,
+        vehicle: Vehicle,
+        date_string: str,
+        trip_period_type: int,
+    ) -> dict:
+        url = self.SPA_API_URL + "vehicles/" + vehicle.id + "/tripinfo"
+        if trip_period_type == 0:  # month
+            if len(date_string) != 6:
+                raise APIError(
+                    f"Invalid date_string, expected yyyymm: {date_string}"
+                )
+            payload = {"tripPeriodType": 0, "setTripMonth": date_string}
+        elif trip_period_type == 1:  # day
+            if len(date_string) != 8:
+                raise APIError(
+                    f"Invalid date_string, expected yyyymmdd: {date_string}"
+                )
+            payload = {"tripPeriodType": 1, "setTripDay": date_string}
+        else:
+            raise APIError(f"Invalid trip period type: {trip_period_type}")
+
+        _LOGGER.debug(f"{DOMAIN} - get_trip_info Request {payload}")
+        response = requests.post(
+            url,
+            json=payload,
+            headers=self._get_authenticated_headers(token),
+        )
+        response = response.json()
+        _LOGGER.debug(f"{DOMAIN} - get_trip_info response {response}")
+        return response
```

Comment:
I suggest calling `_check_response_for_errors` here (same for the other new endpoint), it's a method to check for standard errors in the response